wasm-test-app/src/WASMTests.tsx

```diff
@@ -0,0 +1,212 @@
+import { useState, useEffect } from "react";
+import GenPK from "./GenPk";
+import GenVk from "./GenVk";
+import GenProof from "./GenProof";
+import Verify from './Verify';
+import Hash from './Hash';
+import init, { 
+    gen_pk_wasm, 
+    gen_vk_wasm, 
+    prove_wasm,
+    verify_wasm,
+    poseidon_hash_wasm
+ } from './pkg/ezkl';
+import { readUploadedFileAsText } from './Utils';
+
+const TestScript = () => {
+    const [modelFile, setModelFile] = useState<File | null>(null);
+    const [srsFile, setSrsFile] = useState<File | null>(null);
+    const [circuitSettingsFile, setCircuitSettingsFile] = useState<File | null>(null);
+    const [pkFile, setPkFile] = useState<File | null>(null);
+    const [dataFile, setDataFile] = useState<File | null>(null);
+    const [pkResult, setPkResult] = useState<string | null>(null);
+    const [vkResult, setVkResult] = useState<string | null>(null);
+    const [proofResult, setProofResult] = useState<string | null>(null);
+    const [proofFile, setProofFile] = useState<File | null>(null);
+    const [vkFile, setVkFile] = useState<File | null>(null);
+    const [verifyResult, setVerifyResult] = useState<string | null>(null);
+    const [messageFile, setMessageFile] = useState<File | null>(null);
+    const [hashResult, setHashResult] = useState<string | null>(null);
+
+    // Similar to componentDidMount and componentDidUpdate:
+    useEffect(() => {
+        async function run() {
+        // Initialize the WASM module
+        await init();
+        }
+        run();
+    });
+      
+
+    // Load files
+    useEffect(() => {
+        fetch('/data/test.onnx')
+            .then(res => res.blob())
+            .then(blob => setModelFile(new File([blob], 'test.onnx')));
+
+        fetch('/data/kzg')
+            .then(res => res.blob())
+            .then(blob => setSrsFile(new File([blob], 'kzg')));
+
+        fetch('/data/settings.json')
+            .then(res => res.blob())
+            .then(blob => setCircuitSettingsFile(new File([blob], 'settings.json')));
+
+        fetch('/data/test.provekey')
+            .then(res => res.blob())
+            .then(blob => setPkFile(new File([blob], 'test.provekey')));
+
+        fetch('/data/test.witness.json') // replace 'data' with your actual file name
+            .then(res => res.blob())
+            .then(blob => setDataFile(new File([blob], 'test.witness.json')));
+
+        fetch('/data/test.proof')
+            .then(res => res.blob())
+            .then(blob => setProofFile(new File([blob], 'test.proof')));
+
+        fetch('/data/test.key')
+            .then(res => res.blob())
+            .then(blob => setVkFile(new File([blob], 'test.key')));
+
+        fetch('/data/message')
+            .then(res => res.blob())
+            .then(blob => setMessageFile(new File([blob], 'message')));
+    }, []);
```

Comment:
This is a very large `useEffect` with a lot of ops. Stuff like this screams out to me asking to be placed in another file with a clear name. This practice makes your code more legible and maintainable.